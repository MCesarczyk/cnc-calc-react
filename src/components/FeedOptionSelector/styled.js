import styled from "styled-components";

export const FormLabel = styled.label`
    display: flex;
    gap: 5px;
    align-items: center;
    justify-content: space-between;

    @media (max-width: 480px) {
        flex-direction: column;
        justify-content: center;
    }
`;

export const RadioButtonContainer = styled.div`
    display: flex;
    gap: 10px;
`;