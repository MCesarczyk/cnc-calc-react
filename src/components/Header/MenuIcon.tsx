import { usePathname } from 'next/navigation';
import styled from 'styled-components';

import { ReactComponent as MenuThumbnail } from './hamburger-menu.svg';

interface MenuIconProps {
  onClick: () => void;
}

export const MenuIcon = ({ onClick }: MenuIconProps) => {
  const pathname = usePathname();

  return (
    <MenuIconWrapper onClick={onClick} disabled={pathname === '/'}>
      <Thumbnail />
    </MenuIconWrapper>
  );
};

const Thumbnail = styled(MenuThumbnail)`
  width: 36px;
  margin: -14px 0 0 -14px;
`;

export const MenuIconWrapper = styled.button`
  height: 1.5rem;
  margin: 0.5rem;
  margin-right: 1rem;
  background: transparent;
  border: none;
  color: inherit;
  display: none;
  z-index: 25;
  padding: 20px;

  &:hover {
    filter: brightness(0.8);
  }

  &:active {
    filter: unset;
  }

  &:focus {
    outline: none;
  }

  &:disabled {
    cursor: unset;
    filter: unset;
  }

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    display: block;
    margin: 0.5rem;
  }

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    margin: 0;
  }
`;
