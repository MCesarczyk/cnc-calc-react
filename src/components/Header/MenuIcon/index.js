import { ReactComponent as MenuThumbnail } from "../../../assets/images/hamburger-menu.svg";
import { MenuIconWrapper } from "./styled";

const MenuIcon = ({ onClick }) => (
  <MenuIconWrapper
    onClick={onClick}
  >
    <MenuThumbnail />
  </MenuIconWrapper>

);

export default MenuIcon;