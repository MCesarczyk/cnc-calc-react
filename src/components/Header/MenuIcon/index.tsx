import { useLocation } from 'react-router-dom';
import { ReactComponent as MenuThumbnail } from "../../../assets/images/hamburger-menu.svg";
import { MenuIconWrapper } from './styled';

interface MenuIconProps {
  onClick: () => void;
}

const MenuIcon = ({ onClick }: MenuIconProps) => {
  const { pathname } = useLocation();

  return (
    <MenuIconWrapper onClick={onClick} disabled={pathname === '/'}>
      <MenuThumbnail />
    </MenuIconWrapper>
  );
};

export default MenuIcon;
