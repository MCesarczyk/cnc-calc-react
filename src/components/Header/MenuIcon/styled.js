import styled from "styled-components";

export const MenuIconWrapper = styled.button`
  height: 1.5rem;
  margin: 0.5rem;
  margin-right: 1rem;
  background: transparent;
  border: none;
  color: inherit;
  display: none;
  z-index: 25;

  &:hover {
    filter: brightness(0.8);
  }

  &:active {
    filter: unset;
  }

  &:focus {
    outline: none;
  }

  &:disabled {
    cursor: unset;
    filter: unset;
  }

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    display: block;
    margin: 0.5rem;
  }

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    margin: 0;
  }
`;