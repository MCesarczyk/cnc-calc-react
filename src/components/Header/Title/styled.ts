import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const HomeLink = styled(Link)`
  &:hover {
    filter: brightness(0.8);
  }

  &:active {
    filter: unset;
  }

  &:focus {
    outline: none;
  }
`;

export const Headline = styled.h1`
  font-size: 2rem;
  font-weight: 300;
  font-family: 'Ubuntu', sans-serif;
  text-align: center;
  margin: 0;

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    font-size: 1.5rem;
  }
`;
