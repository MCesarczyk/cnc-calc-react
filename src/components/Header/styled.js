import { Link } from "react-router-dom";
import styled from "styled-components";

export const AppBar = styled.div`
    color: ${({ theme }) => theme.color.primaryContrast};
    background-color: ${({ theme }) => theme.color.primary};
    border-radius: 1.5rem;
    padding: 0.5rem 1.5rem;
    margin: 0 auto 10px auto;
    display: flex;
    align-items: center;
    justify-content: space-between;

    @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
        border-radius: 0;
        margin: 0 0 10px 0;
        padding: 0.5rem;
    }
`;

export const HomeLink = styled(Link)`
    &:hover {
        filter: brightness(0.8);
    }

    &:active {
        filter: unset;
    }

    &:focus {
        outline: none;
    }
`;

export const Headline = styled.h1`
    font-size: 2rem;
    font-weight: 300;
    font-family: 'Ubuntu', sans-serif;
    text-align: center;
    margin: 0;

    @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
        font-size: 1.5rem;
    }
`;