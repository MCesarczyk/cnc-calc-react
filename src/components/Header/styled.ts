import styled from 'styled-components';

export const AppBar = styled.div`
  color: ${({ theme }) => theme.color.primaryContrast};
  background-color: ${({ theme }) => theme.color.primary};
  border-radius: 1.5rem;
  padding: 0.5rem 1.5rem;
  margin: 0 auto 10px auto;
  display: flex;
  align-items: center;
  justify-content: space-between;

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    border-radius: 0;
    margin: 0 0 10px 0;
    padding: 0.5rem;
  }
`;
