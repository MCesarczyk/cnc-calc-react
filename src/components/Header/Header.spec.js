import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";
import { BrowserRouter } from "react-router-dom";
import "jest-styled-components";
import { MenuIconWrapper } from "./MenuIcon/styled";
import { AppBar, Headline, HomeLink } from "./styled";
import MenuIcon from "./MenuIcon";

test("Header rendered properly", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme} >
      <BrowserRouter>
        <AppBar>
          <MenuIcon />
          <HomeLink to="/">
            <Headline />
          </HomeLink>
        </AppBar>
      </BrowserRouter>
    </ThemeProvider>
  );

  let header = component.toJSON();
  expect(header).toMatchSnapshot();
});

test("MenuIcon changes style on action", () => {
  const iconComponent = renderer.create(
    <ThemeProvider theme={theme} >
      <MenuIconWrapper />
    </ThemeProvider>
  );

  let icon = iconComponent.toJSON();

  expect(icon).toHaveStyleRule('filter', 'brightness(0.8)', {
    modifier: ':hover'
  });

  expect(icon).toHaveStyleRule('filter', 'unset', {
    modifier: ':active'
  });

  expect(icon).toHaveStyleRule('outline', 'none', {
    modifier: ':focus'
  });
});

test("Home link changes style on action", () => {
  const homeLinkComponent = renderer.create(
    <ThemeProvider theme={theme} >
      <MenuIconWrapper />
    </ThemeProvider>
  );

  let homeLink = homeLinkComponent.toJSON();

  expect(homeLink).toHaveStyleRule('filter', 'brightness(0.8)', {
    modifier: ':hover'
  });

  expect(homeLink).toHaveStyleRule('filter', 'unset', {
    modifier: ':active'
  });

  expect(homeLink).toHaveStyleRule('outline', 'none', {
    modifier: ':focus'
  });
});