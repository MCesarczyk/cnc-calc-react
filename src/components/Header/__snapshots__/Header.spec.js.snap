// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header rendered properly 1`] = `
.c1 {
  height: 1.5rem;
  margin: 0.5rem;
  margin-right: 1rem;
  background: transparent;
  border: none;
  color: inherit;
  display: none;
  z-index: 25;
}

.c1:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.c1:active {
  -webkit-filter: unset;
  filter: unset;
}

.c1:focus {
  outline: none;
}

.c0 {
  color: #fff;
  background-color: #1e3246;
  border-radius: 1.5rem;
  padding: 0.5rem 1.5rem;
  margin: 0 auto 10px auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.c2:active {
  -webkit-filter: unset;
  filter: unset;
}

.c2:focus {
  outline: none;
}

.c3 {
  font-size: 2rem;
  font-weight: 300;
  font-family: 'Ubuntu',sans-serif;
  text-align: center;
  margin: 0;
}

@media (max-width:768px) {
  .c1 {
    display: block;
    margin: 0.5rem;
  }
}

@media (max-width:480px) {
  .c1 {
    margin: 0;
  }
}

@media (max-width:768px) {
  .c0 {
    border-radius: 0;
    margin: 0 0 10px 0;
    padding: 0.5rem;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 1.5rem;
  }
}

<div
  className="c0"
>
  <button
    className="c1"
  >
    <svg>
      hamburger-menu.svg
    </svg>
  </button>
  <a
    className="c2"
    href="/"
    onClick={[Function]}
  >
    <h1
      className="c3"
    />
  </a>
</div>
`;
